hail/python/hailtop/batch/batch_pool_executor.py

```diff
@@ -511,7 +534,11 @@ async def async_result(self, timeout: Optional[Union[float, int]] = None):
         """
         if self.cancelled():
             raise concurrent.futures.CancelledError()
-        return await asyncio.wait_for(self.fetch_coro, timeout=timeout)
+        try:
+            return await asyncio.wait_for(self.fetch_coro, timeout=timeout)
+        except asyncio.TimeoutError:
+            await self.batch.cancel()
```

Comment:
I think we should call self.async_cancel. Otherwise, fetch_coro will stick around and probably encounter an exception when it realizes the batch is cancelled.